cluster-autoscaler/cloudprovider/aws/README.md

```diff
@@ -175,6 +175,21 @@ If you'd like to scale node groups from 0, an `autoscaling:DescribeLaunchConfigu
 }
 ```
 
+## Using AutoScalingGroup MixedInstancesPolicy
+
+It is possible to use Cluster Autoscaler with a [mixed instances policy](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-autoscaling-autoscalinggroup-mixedinstancespolicy.html), to enable diversification across on-demand and spot instances, of multiple instance types in a single ASG. When using spot instances, this increases the likelihood of successfully launching a spot instance to add the desired capacity to the cluster versus a single instance type, which may be in short supply.
+
+Note that the instance types should have the same amount of RAM and number of CPU cores, since these are the criteria used by CA when scaling.
```

Comment:
This are not the only criteria, just the most commonly used ones. Fundamentally, CA works by assuming every node in ASG is identical in every way. Any difference between the nodes is potentially unsafe. 
If you use nodes with the same cpu/memory/gpu and don't use complex scheduling features (ex. nodeAffinity on system labels) you're _probably_ safe. 
No one is going to guarantee you're ever 100% safe when using mixed ASGs though - it all depends on the detail of how your cluster is configured. For every setup with mixed ASGs you can come up with a pod spec that would lead to autoscaler doing something bad.

Reply:
Thanks for the details. @drewhemm Could you help update to doc?

Reply:
@MaciekPytel Seems we have lots of use cases now and any discussion going on for this topic? I'd like to know anything we can do to make logic not violate core assumption of CA?

Reply:
I believe the only impacted functionality relates to `nodeSelector`, (which will eventually be deprecated) and node affinity and anti-affinity:

https://kubernetes.io/docs/concepts/configuration/assign-pod-node/

I have updated the readme to reflect this.